app/templates/bulk_resource/upload.html

```diff
@@ -18,160 +18,256 @@ <h2 class="ui header">
                 <div class="field">
                     <label for="csv">CSV File</label>
-                    <input id="csv-file" name="csv" placeholder="CSV File" type="file">
+                    <!-- only allow .csv file types -->
+                    <input id="csv-file" name="csv" placeholder="CSV File" type="file" accept=".csv">
                 </div>
                 <div class="inline fields">
                   <div class="field">
-                      <div class="ui radio checkbox">
-                        <input type="radio" id="update" name="reset-or-update" tabindex="0" class="hidden">
-                        <label>Update</label>
-                      </div>
+                    <div class="ui radio checkbox">
+                      <input type="radio" id="reset" name="reset-or-update" checked="" tabindex="0" class="hidden">
+                      <label>Reset</label>
                     </div>
-                    <div class="field">
-                      <div class="ui radio checkbox">
-                        <input type="radio" id="reset" name="reset-or-update" checked="" tabindex="0" class="hidden">
-                        <label>Reset</label>
-                      </div>
+                  </div>
+                  <div class="field">
+                    <div class="ui radio checkbox">
+                      <input type="radio" id="update" name="reset-or-update" tabindex="0" class="hidden">
+                      <label>Update</label>
                     </div>
                   </div>
-                <button class="ui button" type="submit">Upload</button>
-            </form>
-        </div>
-    </div>
-
-    <div class="ui modal">
-        <div class="header">
-            Your CSV file has some errors
-        </div>
-        <div class="image content">
-            <div class="description">
-                <div id="errors">
                 </div>
-            </div>
-        </div>
-        <div class="actions">
-            <div class="ui right labeled button" id="close-button">
-                Got it!
+                <button class="ui disabled button" type="submit" id="upload-button">Upload</button>
+            </form>
+            <h3 class="ui header">Upload Progress</h3>
+            <div class="ui horizontal segments" id="progress-msgs">
+              <div class="ui segment">
+                <h5 class="ui header green">Success Messages</h5>
+                <div class="ui list" id="status-success"></div>
+              </div>
+              <div class="ui segment">
+                <h5 class="ui header red">Error Messages</h5>
+                <div class="ui list" id="status-errors"></div>
+              </div>
             </div>
         </div>
     </div>
 
-    <script src="http://maps.google.com/maps/api/js?sensor=false" type="text/javascript"></script>
-    <script type="text/javascript">
-        $('.ui.radio.checkbox').checkbox();
+  <script type="text/javascript">
```

Comment:
It would be good style to put this in a separate file. it can be placed in a separate file as is I believe as long as it is loaded after `jquery`